### PR TITLE
fix: partially exclude fot from fee

### PR DESCRIPTION
**Pull request by jsy1218**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Bug fix

- **What is the current behavior?** (You can also link to an open issue here)
We exclude all the FOT output token from taking the portion.

- **What is the new behavior (if this is a feature change)?**
We want to change to exclude FOT output token from taking the portion, if the FOT output token:
1. is a transfer to the external address that is not part of uniswap v2 address
2. has a double transfer FOT fee, when going through universal router custody

Both cases are handled by the updated on-chain FOT detector, so in SOR side, we just have to interweave the logic to the portion FOT exclusion part.

- **Other information**:

### CODE REVIEW

**Comment by jsy1218** [comment]
* **#634** <a href="https://app.graphite.dev/github/pr/Uniswap/smart-order-router/634?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jsy1218 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/providers/token-properties-provider.ts [hunk 1]
```diff
@@ -56,8 +56,7 @@ export class TokenPropertiesProvider implements ITokenPropertiesProvider {
     const tokenToResult: TokenPropertiesMap = {};
 
     if (
-      !providerConfig?.enableFeeOnTransferFeeFetching ||
-      this.chainId !== ChainId.MAINNET
```
Comment (xrsv) [inline src/providers/token-properties-provider.ts:60]: this is now using GQL (with onChain fallback) - i believe GQL now supports all chains, correct?

File: src/providers/token-fee-fetcher.ts [hunk 2]
```diff
@@ -27,31 +27,34 @@ type Address = string;
 export type TokenFeeResult = {
   buyFeeBps?: BigNumber;
   sellFeeBps?: BigNumber;
+  feeTakenOnTransfer?: boolean;
```
Comment (xrsv) [inline src/providers/token-fee-fetcher.ts:30]: you'll have to update GQL token fee fetcher to retrieve and expose those fields as well
see [here](https://github.com/Uniswap/routing-api/blob/19b8e7a9e591718233c056eb18a8abb211c61023/lib/graphql/graphql-token-fee-fetcher.ts#L49) and [here](https://github.com/Uniswap/routing-api/blob/19b8e7a9e591718233c056eb18a8abb211c61023/lib/graphql/graphql-queries.ts#L2)

**Review by xrsv — COMMENTED** [review]
looks good by it probably needs some changes in routing api gql fetcher

File: src/providers/token-properties-provider.ts (hunk 1 above)
Comment (jsy1218) [reply src/providers/token-properties-provider.ts:60]: exactly. it's in case GQL fallback

**Pull request merged**